andela/Ah-backend-aquaman
# #164069242 users should be able to view their reading stats

## What does this PR do?

This PR adds a feature that enables a user to view the number of times an article has been read and also view their reading stats

## Description of the task to be completed

- Enable a user to view the number of times an article has been read
- Enable a user to view the number of times they have read a particular article
- Enable a user to view the number of articles they have read

## How should this be manually tested?

- Clone this repository
- cd into the directory Ah-backend-aquaman
- Check out the branch ft-reading-stats-164069242 
- Create and activate your virtual environment
- pip install -r requirements.txt
- python manage.py makemigrations
- python manage.py migrate
- Source your .env file
- python manage.py runserver
- Sign up, verify your email and login
- To read an article go to the endpoint and use the article slug to get the article
http://127.0.0.1:8000/api/articles/articleslug
- To view your read stats go to the endpoint http://127.0.0.1:8000/api/articleslug/readstats/ 

## What are the relevant pivotal tracker stories?

[#164069242](https://www.pivotaltracker.com/story/show/164069242)

## Screenshots
When a user gets an article, they can see the number of people who have read that article as shown

![image](https://user-images.githubusercontent.com/42435346/54788078-e8448a00-4c3e-11e9-9b30-cbb25cd131dd.png)

If a user wishes to view their read stats for a particular article, they go to the endpoint below and using the article slug, are able to get that article, see the number of times they have read it and the number of articles they have read in total

![image](https://user-images.githubusercontent.com/42435346/54805257-7a708080-4c87-11e9-907f-56b1f6d5571c.png)
